server/mqtt.go

```diff
@@ -4463,8 +4557,8 @@ func (sess *mqttSession) cleanupFailedSub(c *client, sub *subscription, cc *Cons
 // Make sure we are set up to deliver PUBREL messages to this QoS2-subscribed
 // session.
 //
-// Session lock held on entry. Need to make sure no other subscribe packet races
-// to do the same.
+// Locks the session for the entire duration. There is no deadlock potential
```

Comment:
The comment here does not match the code. You are not locking for the entire duration...

Reply:
Right, removed. The code is right.